openeo/rest/auth/testing.py

```diff
@@ -151,15 +159,24 @@ def token_callback_resource_owner_password_credentials(self, params: dict, conte
         assert params["scope"] == self.expected_fields["scope"]
         return self._build_token_response()
 
+    def _set_user_code(self):
+        self.state["user_code"] = random_string(length=6).upper()
+
     def device_code_callback(
         self, request: requests_mock.request._RequestObjectProxy, context
     ):
         params = self._get_query_params(query=request.text)
         assert params["client_id"] == self.expected_client_id
         assert params["scope"] == self.expected_fields["scope"]
         self.state["device_code"] = random_string()
-        self.state["user_code"] = random_string(length=6).upper()
         self.state["scope"] = params["scope"]
+
+        # When the mock should also provide the verification_uri_complete then we
+        # we have already set the user_code in the constructor, because we set up
+        # and extra request_mock there for a URL that includes the user_code.
+        if not self.use_verification_uri_complete:
+            self._set_user_code()
```

Comment:
as noted above: the user code should be generated in all cases (with and without use_verification_uri_complete)

Reply:
I agree. Was not ideal and was only needed for the mock that is now removed.